tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -478,6 +547,24 @@ private[scalajs] final class ScalaJSClassEmitter(
     }
   }
 
+  private[tools] def genIntervalsTest(className: String, tag: js.Tree)
+      (implicit pos: Position): js.Tree = {
+    import TreeDSL._
+
+    val intervals = subtypeIntervals(className)
+
+    if(intervals.nonEmpty && !needsSubtypeArray(className)){
+      intervals.map {
+        case (a, b) if a == b => tag === js.IntLiteral(a)
+        case (lo, hi) =>
+          js.BinaryOp(JSBinaryOp.>=, tag, js.IntLiteral(lo)) &&
+          js.BinaryOp(JSBinaryOp.<=, tag, js.IntLiteral(hi))
+      } reduce(_ || _)
+    } else {
+      js.BracketSelect(envField("Is", className), tag)
```

Comment:
Hum ... What happens when `intervals.isEmpty && !needsSubtypeArray(className)`? This seems like it would try to access an `Is_xyz` that is not declared. What does it mean when `intervals.isEmpty`?